website/versioned_docs/version-v1.x/contracts.md

```diff
@@ -229,68 +251,111 @@ emptyBallotRoots[3] = uint256(12031563002271722465187541954825013132282571927669
+
+```ts
+function deploy(
+  uint256 _duration,
+  MaxValues calldata _maxValues,
+  TreeDepths calldata _treeDepths,
+  PubKey calldata _coordinatorPubKey,
+  address _maci,
+  TopupCredit _topupCredit,
+  address _pollOwner
+) public virtual returns (address pollAddr) {
+  /// @notice Validate _maxValues
+  /// maxVoteOptions must be less than 2 ** 50 due to circuit limitations;
+  /// it will be packed as a 50-bit value along with other values as one
+  /// of the inputs (aka packedVal)
+  if (_maxValues.maxVoteOptions >= (2 ** 50)) {
+    revert InvalidMaxValues();
+  }
+
+  /// @notice deploy a new AccQueue contract to store messages
+  AccQueue messageAq = new AccQueueQuinaryMaci(_treeDepths.messageTreeSubDepth);
+
+  /// @notice the smart contracts that a Poll would interact with
+  ExtContracts memory extContracts = ExtContracts({
+    maci: IMACI(_maci),
+    messageAq: messageAq,
+    topupCredit: _topupCredit
+  });
+
+  // deploy the poll
+  Poll poll = new Poll(_duration, _maxValues, _treeDepths, _coordinatorPubKey, extContracts);
+
+  // Make the Poll contract own the messageAq contract, so only it can
+  // run enqueue/merge
+  messageAq.transferOwnership(address(poll));
+
+  // init Poll
+  poll.init();
+
+  poll.transferOwnership(_pollOwner);
+
+  pollAddr = address(poll);
+}
 ```
 
-Upon deployment, the ownership of the messageAq contract will be transferred to the deployed poll, as well as the ownership of the new Poll contract be transferred to the poll owner, which in MACI is set as the owner of MACI.
+Upon deployment, the following will happen:
+
+- ownership of the `messageAq` contract is transferred to the deployed poll contract
+- ownership of the new `Poll` contract is transferred to the poll owner, which in `MACI` is set as the owner of `MACI`.
 
 ## MessageProcessor
 
-This contract is used to prepare parameters for the zk-SNARK circuits as well as for verifying proofs. It should be deployed alongside MACI and ownership assigned to the coordinator.
-It will process messages in batches, to increase performance and stay within the block gas limit.
-After it finishes processing, the sbCommitment will be used for Tally and Subsidy contracts.
+This contract is used to prepare parameters for the zk-SNARK circuits as well as for verifying proofs. It should be deployed alongside `MACI` and ownership assigned to the coordinator.
+It will process messages in batches, and after all batches have been processed, the `sbCommitment` can then be used for the `Tally` and `Subsidy` contracts.
 
 ## Tally
 
-The Tally contract is used during votes tallying and by users to verify the tally results.
+::info
+The `Tally` contract is present also in a non quadratic voting fashion, and is slightly smaller due to not having the `verifyPerVOSpentVoiceCredits` function. This is not required because with normal (non quadratic) voting, each vote by a user is not the square root of the voice credits spent.  
```

Comment:
is the **fashion** here be the **function**?

Reply:
no, it's meant to be fashion. Does it not make sense?

Reply:
Not sure hhhh since my english is also poor. Either is fine.